kadena-community/kadena.js
# Add Integration test for transaction subscriptions

### Add Integration test for transaction subscriptions

Related Issue/Asana ticket: [Asana Ticket](https://app.asana.com/0/1206316065856327/1207378878709844/f)

Short description: 
Adds an integration test for subscribing to transactions using GraphQL. This ensures that the various states a transaction can be in are correctly returned when using subscriptions.

### Test scenarios
- Create a transaction, execute this up until `preflight`  using client utils `executeTo()` function.  This yields the `requestKey`
- Set up the subscription using `graphql-ws` (tried natively with Playwright, unfortunately only HTTP requests are supported)
- Assert that the first and second response contain the expected values

### Reminders (if applicable)

- [X] I ran `pnpm install` and `pnpm test` in the root of the monorepo
      (optionally with `--filter=...package...` to exclude non-affected
      projects)
- [/] I ran `pnpm changeset` in the root of the monorepo
- [X] Test coverage has not decreased

### Questions
In order to do this test, I needed full control of the transaction and it's state. The `executeTo()` function provides this. This function results in a `task` that you can then proceed with (at a later stage) I wanted to move this logic to a separate helperfile, but I got stuck on the typings of the task. It might be a nice improvement to look at this together with one of you.
